Feature/modularisation #167

jannahastings opened the pull request:
As per the proposal presented in issue #167, the ontology file has been split into three editing modules: oeo-physical.omn, oeo-social.omn and oeo-model.omn. The primary ontology file oeo.omn now imports those modules. All the entities should appear the same, however, with the exception of the following substantial changes that were made during the modularisation: 

- Added a description of ontology and each submodule as an ontology annotations on each module file and the main importing file
- Removed definition of “geothermal generator” that was annotated to class GitHub and added it to the class GeothermalGenerator”
- Removed CO2 type “Cost”, added CO2 type GreenhouseGas. 
- Changed class name “GeopgraphicCoordinate” to “GeographicCoordinate”
- Classes Bus, Generator, PowerGeneratingUnit etc. (from physical module) no longer explicitly disjoint with Consumer (social module) however they are pairwise disjoint with each other
- Publication (social module) no longer subclass of Reference (model module)
- Weblink (model module) no longer subclass of Link (physical module)
- Added myself as a contributor

Note that the object properties are defined in the modules in which they are used. Those that are used in multiple modules are defined in multiple modules, however, their annotations are added only in one module (or else the annotations would be duplicated). Those object properties that were not used at all are defined in the master file 'oeo.omn'.

Closes #167.

akleinau commented:
The changes got implemented as discussed, the three modules are working and still make sense to me.

Some things I saw:
- Quantity and subclasses are in both physical and social (and should just be physical)
- Graph, Link and Node (and subclasses) are in both physical and social (and should just be physical)
- Month and Year are in physical but  they are also needed for model description and social constructs
- MaStr and subclasses should be deleted because they are subclass of dataset (model) but get very physical (and should've been deleted by now anyway)

Note: my protege had problems to open oeo, this could just be a problem of me but I wanted to note it in case anyone else has problems too

Note 2: When we have a definition for sector we should think about its classification again

akleinau commented:
I added what worked for me to open the oeo.omn file as a commit, please report if there should still be problems. 
I couldnt open the oeo file because the import links in the file dont link to the ontology modules (because they arent released yet)

jannahastings commented:
@akleinau I removed the duplicated classes Quantity, Graph, Link and Node (and all descendents) from oeo-model, and deleted MaStr. I left Month and Year where they are for now, because I think these are also good candidates for importing from an external ontology into a central re-use module (to be investigated, which ontology is a good candidate).
